File: apps/mobile/types/transaction.ts
export type TransactionType = 'income' | 'expense';

export interface Transaction {
  id: number;
  title: string;
  amount: number;
  type: TransactionType;
  category: string;
  description?: string;
  created_at: string;
  updated_at?: string;
}

export interface TransactionCreate {
  title: string;
  amount: number;
  type: TransactionType;
  category: string;
  description?: string;
}

export interface TransactionSummary {
  balance: number;
  total_income: number;
  total_expense: number;
}

File: services/backend/models/transaction.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from core.database import Base

class TransactionModel(Base):
    __tablename__ = "transactions"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(200), index=True, nullable=False)
    amount = Column(Integer) # Stored in cents
    type = Column(String(20), index=True, nullable=False)  # 'income' or 'expense'
    category = Column(String(100), index=True, default="General")
    description = Column(String(500))
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, onupdate=datetime.utcnow)
    
    owner = relationship("UserModel", back_populates="transactions")
